libraries/hatch-server-user-management/src/UserInfoRequest.ts

```diff
@@ -0,0 +1,70 @@
+import {BasicRouteParams} from '@launchtray/hatch-server-middleware';
+import {containerSingleton, Logger} from '@launchtray/hatch-util';
+import cookie from 'cookie';
+import {AUTH_ACCESS_TOKEN_COOKIE_NAME} from './constants';
+import AWSCognitoClient from './AWSCognitoClient';
+import {UserInfo} from './UserServiceClient';
+
+@containerSingleton()
+export default class UserInfoRequest {
+  private userInfo: UserInfo | undefined;
+  private logger: Logger;
+  
+  constructor(public readonly params: BasicRouteParams, private readonly userService: AWSCognitoClient) {
+    this.logger = params.logger;
+  }
+  
+  public async getUserInfo() {
+    if (!this.userInfo) {
+      await this.authenticateUser();
+    }
+    return this.userInfo;
+  }
+  
+  private async authenticateUser() {
+    const token = this.extractAuthenticationCookie() || this.extractAuthenticationHeader();
```

Comment:
Let's use ?? for things like this rather than ||

Reply:
Updated